include/orb/Orbit.h

```diff
@@ -388,5 +393,261 @@ class Orbit {
+        lin::Vector3d r_ecef0= rel_r+orb_r;
+        lin::Matrix<double, 3, 3> dcm_ecef_ecef0;
+        relative_earth_dcm_helper(_earth_rate_ecef, t, dcm_ecef_ecef0);
+        lin::Vector3d pos_ecef= dcm_ecef_ecef0*r_ecef0;
+        double potential;
+        lin::Vector3d g_ecef;
+        calc_geograv(pos_ecef, g_ecef, potential);
+        //convert to ECEF0
+        return (lin::transpose(dcm_ecef_ecef0)*g_ecef + orb_r*_mu_a3).eval();
+    }
+
+    /**
+     * Convert _recef and _vecef back to ecef from relative position and velocity in inertial ecef0.
+     * uses orbit info, _t, _mu_a3, _x, _y, _omega
+     *
+     * grav calls: 0
+     */
+    void _unrelativize_helper(){
+        //convert back to absolute ecef
+        double theta= _t*lin::norm(_omega);
+        double costheta= std::cos(theta);
+        double sintheta= std::sin(theta);
+        lin::Vector3d orb_r= _x*costheta+_y*sintheta;
+        _recef= _recef+ orb_r;
+        _vecef= _vecef+ lin::cross(_omega,orb_r);
+        // rotate back to ecef
+        lin::Matrix<double, 3, 3> dcm_ecef_ecef0;
+        relative_earth_dcm_helper(_earth_rate_ecef, _t, dcm_ecef_ecef0);
+        _recef= (dcm_ecef_ecef0*_recef).eval();
+        _vecef= (dcm_ecef_ecef0*_vecef).eval();
+        // remove cross r term from velocity
+        _vecef= _vecef-lin::cross(_earth_rate_ecef,_recef);
+    }
+
+
+
+
+
+    /**
+     * If propagating call the gravity model once
+     * to move the propagtor forward, otherwise do nothing.
+     * High order integrators from:
+     * https://doi.org/10.1016/0375-9601(90)90092-3
+     *
+     * grav calls: 1 if propagating, 0 if not propagating
+     */
+    void onegravcall(){
+        //high order integrators
+        //https://doi.org/10.1016/0375-9601(90)90092-3
+        static const std::array<double,7> d{{0.784513610477560L,
+                                        0.235573213359357L,
+                                        -1.177679984178870L,
+                                        1.315186320683906L,
+                                        -1.177679984178870L,
+                                        0.235573213359357L,
+                                        0.784513610477560L}};
+        if (!valid()){
+            return;
+        }
+        if (_numgravcallsleft==0){
+            //done propagating
+            return;
+        }
+        //now what dt should be used?
+        double dt;
+        if (_longstep==0){
+            //not inside a long step
+            //convert to relative ecef0
+            _relativize_helper();
+            int64_t deltatime= _targetgpstime-_ns_gps_time;
+            int signofdt= (deltatime<0)?-1:1;
+            int64_t currentdtns;
+            if (_numgravcallsleft>=7){
+                // do a long step
+                if (std::abs(deltatime)>=std::abs(maxlongtimestep)){
+                    // take a full time step
+                    currentdtns= signofdt*maxlongtimestep;
```

Comment:
Again, just to make sure I'm understanding this correctly, this is a 7th(ish) order fixed step size orbit propagator correct?
_Not sure about seventh order but wanna make sure it's fixed step size!_

Reply:
The higher order propagator step right now works like this, first it converts position and velocity in ecef to relative inertial coordinates to a close reference circular orbit. Then it does a series of drift-kick-drift steps (see https://en.wikipedia.org/wiki/Leapfrog_integration ) where a drift is `rel_r= rel_r+rel_v*dt*0.5;` and a kick is `rel_v= rel_v + g_ecef0*dt;` For the low order step(2nd ish) there is just one drift-kick-drift, for the higher order step(6th ish) Yoshida coefficients are used to do 7 drift-kick-drifts with a series of d*dt:
```
static const std::array<double,7> d{{0.784513610477560L,
                                        0.235573213359357L,
                                        -1.177679984178870L,
                                        1.315186320683906L,
                                        -1.177679984178870L,
                                        0.235573213359357L,
                                        0.784513610477560L}};
```
Where somehow this magical series of time steps cause some errors to cancel out. Finally when the step(s) are done the relative position and velocity are converted back to ecef.

Reply:
Yeah no makes sense. But it's fixed step in the sense that there is no adaptive step size mechanism like `ode45` or `ode23`. Not at all saying that I think we need it, just trying to confirm as observation.